### ISSUE TITLE
twisted.conch.test.test_ssh uses some deprecated twisted.conch.ssh.keys APIs

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4408]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#4408|
|Type|defect|
|Created|2010-04-23 14:08:34Z|
|Branch|https://github.com/twisted/twisted/tree/test_ssh-key-updates-4408|

`keys.getPublicKeyString` and `keys.getPrivateKeyObject` have been deprecated for a while, but `test_ssh` still uses them.  It should be using the replacements described in the deprecation warnings emitted when it is run.

<details><summary>Searchable metadata</summary>

```
trac-id__4408 4408
type__defect defect
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_test_ssh_key_updates_4408 branches/test_ssh-key-updates-4408
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__conch conch
keywords__ 
time__1272031714000000 1272031714000000
changetime__1272650162000000 1272650162000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [28850]) Branching to 'test_ssh-key-updates-4408'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

(In [28851]) Convert getPublicKeyString and getPrivateKeyObject calls to the new Key-based APIs

refs [#4408](https://github.com/twisted/twisted/issues/4408)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun removed owner|
|-|-|

Pretty simple change; despite the large number of warnings, there were only a few pieces of code that needed to change.

In trunk, I see 280 deprecation warnings from test_ssh.  In the branch, I see _only_ 93.

Please review.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun commented|
|-|-|

[Build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/test_ssh-key-updates-4408)

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_5"></a>|@itamarst set owner to @exarkun|
|-|-|

1. I'm confused by the diffs in lines 296 and 379: everywhere else with a getPublicKeyString you have a .blob() call on the fromString() result, since that's what the deprecated method does internally. In these case you don't call blob(), and yet tests still pass? How is that?

2. getPrivateKeyObject internally does a getString().keyObject, but deprecation message says to do just do getString(). You follow the deprecation message - is that OK?

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun commented|
|-|-|

> 1. I'm confused by the diffs in lines 296 and 379: everywhere else with a getPublicKeyString you have a .blob() call on the fromString() result, since that's what the deprecated method does internally. In these case you don't call blob(), and yet tests still pass? How is that? 

These lines were actually doing two deprecated things simultaneously.  First, using the deprecated `getPublicKeyString`.  Second, returned public key strings, rather than public key objects, from `getPublicKeys`.

> 2. getPrivateKeyObject internally does a getString().keyObject, but deprecation message says to do just do getString(). You follow the deprecation message - is that OK? 

Yea.  The `keyObject` is a PyCrypto object.  The Conch APIs want Conch Key objects instead.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun set status to `closed`|
|-|-|

(In [28885]) Merge test_ssh-key-updates-4408

Author: exarkun
Reviewer: itamar
Fixes: [#4408](https://github.com/twisted/twisted/issues/4408)

Update the use of several deprecated APIs in the Conch test suite to more
modern APIs.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
